Source/Common/Public/Concurrent/Lock.h

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 Chenxi Xu (@xsun2001)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <mutex>
+#include <shared_mutex>
+#include <condition_variable>
+
+namespace SpaceGameEngine
+{
+	class Mutex
+	{
+	public:
+		Mutex();
+
+		Mutex( const Mutex &other ) = delete;
+
+		~Mutex() = default;
+
+		Mutex &operator=( const Mutex &&other ) = delete;
```

Comment:
four function?only two here!

Reply:
std::recursive_timed_mutex is NOT MoveConstructible and MoveAssignable. Compilers will not generate move constructors when copy constructors exist.